DataFormats/TrackerRecHit2D/interface/VectorHit.h

```diff
@@ -0,0 +1,160 @@
+ *
+ * $Date: 2015/03/30 $
+ * \author Erica Brondolin
+ *
+ */
+
+#include "DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/OmniClusterRef.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrackingRecHit/interface/KfComponentsHolder.h"
+
+#include "TkCloner.h"
+
+class VectorHit : public BaseTrackerRecHit {
+public:
+  typedef OmniClusterRef::Phase2Cluster1DRef ClusterRef;
+
+  VectorHit() : thePosition(), theDirection(), theCovMatrix(), theDimension(0) { setType(bad); }
+
+  VectorHit(const VectorHit& vh);
+
+  VectorHit(const GeomDet& idet,
+            const LocalPoint& posInner,
+            const LocalVector& dir,
+            const AlgebraicSymMatrix& covMatrix,
+            const double& Chi2,
+            OmniClusterRef const& lower,
+            OmniClusterRef const& upper);
+
+  VectorHit(const GeomDet& idet,
+            const VectorHit2D& vh2Dzx,
+            const VectorHit2D& vh2Dzy,
+            OmniClusterRef const& lower,
+            OmniClusterRef const& upper);
+
+  ~VectorHit() override;
+
+  VectorHit* clone() const override { return new VectorHit(*this); }
+#ifndef __GCCXML__
+  RecHitPointer cloneSH() const override { return std::make_shared<VectorHit>(*this); }
+#endif
+
+  bool sharesInput(const TrackingRecHit* other, SharedInputType what) const override;
+  bool sharesClusters(VectorHit const& h1, VectorHit const& h2, SharedInputType what) const;
+
+  // Parameters of the segment, for the track fit
+  // For a 4D segment: (dx/dz,dy/dz,x,y)
+  bool hasPositionAndError() const override {
+    //bool hasPositionAndError() const {
+    return true;
+    //      return (err_.xx() != 0) || (err_.yy() != 0) || (err_.xy() != 0) ||
+    //             (pos_.x()  != 0) || (pos_.y()  != 0) || (pos_.z()  != 0);
+  };
+
+  AlgebraicVector parameters() const override;
+  void getKfComponents(KfComponentsHolder& holder) const override { getKfComponents4D(holder); }
+  void getKfComponents4D(KfComponentsHolder& holder) const;
+
+  // returning methods
+  LocalPoint localPosition() const override { return thePosition; }
+  virtual LocalVector localDirection() const { return theDirection; }
+  AlgebraicSymMatrix parametersError() const override;
+  LocalError localPositionError() const override;
+  virtual LocalError localDirectionError() const;
+  Global3DVector globalDirection() const;
+
+  virtual double chi2() const { return theChi2; }
+  int dimension() const override { return theDimension; }
+
+  std::pair<double, double> curvatureORphi(std::string curvORphi = "curvature") const;
```

Comment:
1) why double?
2) use enumerator, not string

Reply:
Reduced to float, using enum now.